### PR TITLE
Convert HD-Space from csQuery to AngleSharp

### PR DESCRIPTION
**Pull request by cadatoiva**
Tested and working on my account. I left in the old csQuery Code as hints for the other trackers we're trying to convert to remove csQuery. There's almost a 1-to-1 relationship in functions that should make the process fairly easy.

### CODE REVIEW

**Comment by cadatoiva** [comment]
#7292

File: src/Jackett.Common/Indexers/HDSpace.cs [hunk 1]
```diff
@@ -102,90 +96,93 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
-
-            var response = await RequestStringWithCookiesAndRetry(searchUrl);
-            var results = response.Content;
+            var response = await RequestStringWithCookiesAndRetry(SearchUrl + queryCollection.GetQueryString());
 
             try
             {
-                CQ dom = results;
-                var rows = dom["table.lista > tbody > tr"];
+                //TODO remove csQuery -> AngleSharp hints when all trackers are converted
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDSpace.cs]: Please, remove all code related to csQuery. Related changes must be done in the same PR.
I will review all PRs ASAP

File: src/Jackett.Common/Indexers/HDSpace.cs [hunk 2]
```diff
@@ -102,90 +96,93 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
-
-                    var sizeStr = row.ChildElements.ElementAt(5).Cq().Text();
+                    //var dateStr = row.ChildElements.ElementAt(4).Cq().Text().Trim();
+                    var dateStr = row.Children[4].Text().Trim();
+                    //if (dateStr.StartsWith("Today"))
+                    //    release.PublishDate = DateTime.Today + TimeSpan.ParseExact(dateStr.Replace("Today at ", ""), "hh\\:mm\\:ss", CultureInfo.InvariantCulture);
+                    //else if (dateStr.StartsWith("Yesterday"))
+                    //    release.PublishDate = DateTime.Today - TimeSpan.FromDays(1) + TimeSpan.ParseExact(dateStr.Replace("Yesterday at ", ""), "hh\\:mm\\:ss", CultureInfo.InvariantCulture);
+                    //else
+                    //    release.PublishDate = DateTime.SpecifyKind(DateTime.ParseExact(dateStr, "MMMM dd, yyyy, HH:mm:ss", CultureInfo.InvariantCulture), DateTimeKind.Local);
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDSpace.cs]: I assume that `DateTimeUtil.FromUnknown` handles all cases. Is that right?

File: src/Jackett.Common/Indexers/HDSpace.cs [hunk 3]
```diff
@@ -102,90 +96,93 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
                     release.Size = ReleaseInfo.GetBytes(sizeStr);
-
-                    release.Seeders = ParseUtil.CoerceInt(row.ChildElements.ElementAt(7).Cq().Text());
-                    release.Peers = ParseUtil.CoerceInt(row.ChildElements.ElementAt(8).Cq().Text()) + release.Seeders;
-
-                    var grabs = qRow.Find("td:nth-child(10)").Text();
+                    release.Seeders = ParseUtil.CoerceInt(row.Children[7].Text());
+                    release.Peers = ParseUtil.CoerceInt(row.Children[8].Text()) + release.Seeders;
+                    //var grabs = qRow.Find("td:nth-child(10)").Text();
+                    var grabs = row.QuerySelector("td:nth-child(10)").Text();
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDSpace.cs]: Are you sure `row.QuerySelector("td:nth-child(10)")` works as expected? I remember having problems with something similar...

File: src/Jackett.Common/Indexers/HDSpace.cs (hunk 2 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/HDSpace.cs]: Yes, FromUnknown (https://github.com/Jackett/Jackett/blob/28b53d8fb1b6161c69ae5b76e68d6f3b66aedcbd/src/Jackett.Common/Utils/DateTimeUtil.cs#L130) first checks for "today"/"tomorrow", removes it and checks for a valid time. It doesn't check/remove the "at" though, which is why my replacement removes it before passing. If it finds no other special times, it just does a normal parse.

All three cases in this tracker work as expected when I tested with the new code.

File: src/Jackett.Common/Indexers/HDSpace.cs [hunk 4]
```diff
@@ -102,90 +96,80 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
-                    if (dateStr.StartsWith("Today"))
-                        release.PublishDate = DateTime.Today + TimeSpan.ParseExact(dateStr.Replace("Today at ", ""), "hh\\:mm\\:ss", CultureInfo.InvariantCulture);
-                    else if (dateStr.StartsWith("Yesterday"))
-                        release.PublishDate = DateTime.Today - TimeSpan.FromDays(1) + TimeSpan.ParseExact(dateStr.Replace("Yesterday at ", ""), "hh\\:mm\\:ss", CultureInfo.InvariantCulture);
-                    else
-                        release.PublishDate = DateTime.SpecifyKind(DateTime.ParseExact(dateStr, "MMMM dd, yyyy, HH:mm:ss", CultureInfo.InvariantCulture), DateTimeKind.Local);
-
-                    var sizeStr = row.ChildElements.ElementAt(5).Cq().Text();
+                    var dateStr = row.Children[4].Text().Trim();
+                    //if (dateStr.StartsWith("Today"))
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDSpace.cs]: remove the commented code and let few date examples
```
// Today at hh:mm:ss
// Yesterday at hh:mm:ss
// MMMM dd, yyyy, HH:mm:ss
```

File: src/Jackett.Common/Indexers/HDSpace.cs [hunk 5]
```diff
@@ -102,90 +96,80 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
-
-                    var grabs = qRow.Find("td:nth-child(10)").Text();
+                    release.Seeders = ParseUtil.CoerceInt(row.Children[7].Text());
+                    release.Peers = ParseUtil.CoerceInt(row.Children[8].Text()) + release.Seeders;
+                    var grabs = row.QuerySelector("td:nth-child(10)").Text();
                     grabs = grabs.Replace("---", "0");
                     release.Grabs = ParseUtil.CoerceInt(grabs);
-
-                    if (qRow.Find("img[title=\"FreeLeech\"]").Length >= 1)
+                    if (row.QuerySelectorAll("img[title=\"FreeLeech\"]").Length >= 1)
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDSpace.cs]: row.QuerySelector("img[title=\"FreeLeech\"]") != null
There are other cases below

**Pull request merged**